core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -136,8 +139,8 @@ public void onNext(T t) {
 
         @Override
         public void onError(Throwable t) {
-            beforeError(delegate, t);
-            delegate.onError(t);
+            final Throwable filteredCause = beforeError(delegate, t);
+            delegate.onError(filteredCause != null ? filteredCause : t);
```

Comment:
How about translating `null` into `onComplete()` instead of falling back to the original cause?

Reply:
It makes sense. I will check it again.

Reply:
As we talked about this offline, there is no way to make a response if returning `null`. So it would be better not fixing this.